### PR TITLE
sql/sqlite: type registry

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: sql/sqlite/sqlspec.go [hunk 1]
```diff
@@ -275,3 +276,34 @@ func enumSpec(t *schema.EnumType) (*sqlspec.Column, error) {
...
+	specutil.TypeSpec("nchar", specutil.SizeTypeAttr(false)),
+	specutil.AliasTypeSpec("native_character", "native character", specutil.SizeTypeAttr(false)),
+	specutil.TypeSpec("nvarchar", specutil.SizeTypeAttr(false)),
+	specutil.TypeSpec("clob", specutil.SizeTypeAttr(false)),
+	specutil.TypeSpec("numeric", specutil.SizeTypeAttr(false)),
+	specutil.TypeSpec("decimal", &schemaspec.TypeAttr{Name: "precision", Kind: reflect.Int, Required: false}, &schemaspec.TypeAttr{Name: "scale", Kind: reflect.Int, Required: false}),
+	specutil.TypeSpec("boolean"),
+	specutil.TypeSpec("date"),
+	specutil.TypeSpec("datetime"),
+)
```
Comment (a8m) [inline sql/sqlite/sqlspec.go:309]: json and uuid are not real types, but let's add them to support ent.

File: sql/sqlite/sqlspec_test.go [hunk 2]
```diff
@@ -386,3 +390,143 @@ func TestNotSupportedMarshalSpecColumnType(t *testing.T) {
...
+			column := test.Columns[0]
+			typ, err := TypeRegistry.Type(column.Type, column.Extra.Attrs, parseRawType)
+			require.NoError(t, err)
+			require.EqualValues(t, tt.expected, typ)
+			spec, err := hclState.MarshalSpec(&test)
+			require.NoError(t, err)
+			require.EqualValues(t, string(hclwrite.Format([]byte(doc))), string(hclwrite.Format(spec)))
+		})
+	}
+}
```
Comment (a8m) [inline sql/sqlite/sqlspec_test.go:532]: How do you handle empty type?

**Review by a8m — APPROVED** [review]

File: sql/sqlite/sqlspec_test.go [hunk 3]
```diff
@@ -8,14 +8,18 @@ import (
...
 	"testing"
 
+	"ariga.io/atlas/schema/schemaspec"
 	"ariga.io/atlas/schema/schemaspec/schemahcl"
 	"ariga.io/atlas/sql/internal/specutil"
 	"ariga.io/atlas/sql/schema"
 	"ariga.io/atlas/sql/sqlspec"
+	"github.com/hashicorp/hcl/v2/hclwrite"
 
 	"github.com/stretchr/testify/require"
```
Comment (a8m) [inline sql/sqlite/sqlspec_test.go:18]:
```suggestion

	"github.com/hashicorp/hcl/v2/hclwrite"
	"github.com/stretchr/testify/require"
```

File: sql/sqlite/sqlspec_test.go [hunk 4]
```diff
@@ -386,3 +390,143 @@ func TestNotSupportedMarshalSpecColumnType(t *testing.T) {
...
+			typeExpr: "date",
+			expected: &schema.TimeType{T: "date"},
+		},
+		{
+			typeExpr: "datetime",
+			expected: &schema.TimeType{T: "datetime"},
+		},
+	} {
+		t.Run(tt.typeExpr, func(t *testing.T) {
+			// simulates sqlspec.Column until we change its Type field.
```
Comment (a8m) [inline sql/sqlite/sqlspec_test.go:509]:
```suggestion
			// Simulates sqlspec.Column until we change its Type field.
```

File: sql/sqlite/sqlspec_test.go (hunk 2 above)
Comment (rotemtam) [reply sql/sqlite/sqlspec_test.go:532]: on a higher level where the `type` field in the column isn't set. 
I'll add a test for it during the refac

**Pull request merged**